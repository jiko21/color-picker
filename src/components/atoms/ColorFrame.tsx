import { Flex, View } from '@adobe/react-spectrum';
import React from 'react';

interface Props {
  children: React.ReactNode;
  color: string;
  width?: string;
  height?: string;
}

export const ColorFrame: React.FC<Props> = React.memo(({ children, color, width, height }) => (
  <Flex
    UNSAFE_style={{
      backgroundColor: color,
    }}
    width={width}
    height={height}
    alignItems="center"
    justifyContent="center"
  >
    <View>{children}</View>
  </Flex>
));
