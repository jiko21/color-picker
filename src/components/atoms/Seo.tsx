import Head from 'next/head'
import React from 'react';

interface SeoProps {
  colors: string[];
}
export const Seo: React.FC<SeoProps> = ({ colors }) => {
  const ogpUrl = React.useMemo( () => {
    if (colors.length > 0) {
      const urlSearchParams = new URLSearchParams();
      colors.forEach((color) => {
        urlSearchParams.append('color', color);
      });
      return `https://color-picker-jiko21.vercel.app/api/og?${urlSearchParams.toString()}`;
    } else {
      return 'https://color-picker-jiko21.vercel.app/api/og';
    }
  }, [colors]);

  return (
    <Head>
      <title>color picker</title>
      <link rel="apple-touch-icon" sizes="180x180" href="favicon/apple-touch-icon.png" />
      <link rel="icon" type="image/png" sizes="32x32" href="favicon/favicon-32x32.png" />
      <link rel="icon" type="image/png" sizes="16x16" href="favicon/favicon-16x16.png" />
      <link rel="manifest" href="favicon/site.webmanifest" />
      <meta name="description" content="color selegct app" />
      <meta property="og:title" content="color picker" />
      <meta property="og:type" content="website" />
      <meta property="og:url" content="https://color-picker-jiko21.vercel.app" />
      <meta property="og:image" content={ogpUrl} />
      <meta property="og:site_name" content="Colory" />
      <meta property="og:description" content="color select app" />
      <meta name="twitter:card" content="summary" />
      <meta name="twitter:site" content="@jiko_21" />
    </Head>
  )
};
