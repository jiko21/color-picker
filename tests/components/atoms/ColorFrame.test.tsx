import { ColorFrame } from '@/components/atoms/ColorFrame';
import { render } from '@testing-library/react';
import React from 'react';

describe('@/components/atoms/ColorFrame.tsx', () => {
  it('correctly render', () => {
    const component = render(<ColorFrame color={'#ff0000'}>children</ColorFrame>);
    expect(component.container).toMatchSnapshot();
  });
});
