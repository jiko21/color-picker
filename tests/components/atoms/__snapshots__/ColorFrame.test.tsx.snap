// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@/components/atoms/ColorFrame.tsx correctly render 1`] = `
<div>
  <div
    class="flex_e15493"
    style="background-color: rgb(255, 0, 0); align-items: center; justify-content: center;"
  >
    <div>
      children
    </div>
  </div>
</div>
`;

exports[`components/Atoms/ColorFrame.tsx correctly render 1`] = `
<div>
  <div
    class="flex_e15493"
    style="background-color: rgb(255, 0, 0); align-items: center; justify-content: center;"
  >
    <div>
      children
    </div>
  </div>
</div>
`;
